website/common/script/libs/getItemInfo.js

```diff
@@ -9,12 +9,25 @@ import isFreeRebirth from './isFreeRebirth';
 import getOfficialPinnedItems from './getOfficialPinnedItems';
 
 function lockQuest (quest, user) {
-  if (quest.key === 'lostMasterclasser1') return !(user.achievements.quests.dilatoryDistress3 && user.achievements.quests.mayhemMistiflying3 && user.achievements.quests.stoikalmCalamity3 && user.achievements.quests.taskwoodsTerror3);
-  if (quest.lvl && user.stats.lvl < quest.lvl) return true;
-  if (quest.unlockCondition && (quest.key === 'moon1' || quest.key === 'moon2' || quest.key === 'moon3')) {
-    return user.loginIncentives < quest.unlockCondition.incentiveThreshold;
+  // checks series quests, including Masterclasser
+  if (quest.prereqQuests) {
+    if (!user.achievements.quests) return true;
+    for (const prereq of quest.prereqQuests) {
+      if (!user.achievements.quests[prereq]) return true;
+    }
   }
-  if (user.achievements.quests) return quest.previous && !user.achievements.quests[quest.previous];
+  // checks quest & user level against quest level
+  if (quest.lvl && user.stats.lvl < quest.lvl) return true;
+
+  // checks unlockCondition.incentiveThreshold
+  if (
+    quest.unlockCondition
+    && quest.unlockCondition.incentiveThreshold
+    && user.loginIncentives < quest.unlockCondition.incentiveThreshold
+  ) return true;
+  if (user.achievements.quests
+    && !user.achievements.quests[quest.previous]);
+  // then if we've passed all the checks
   return quest.previous;
```

Comment:
this last part of the logic has me puzzled--
what's with the `if` statement that just ends in a semicolon?
and why is the final return still `quest.previous`?

I'm guessing it should be

```
if (quest.previous) return !(user.achievements.quests && user.achievements.quests[quest.previous]);
return false;
```

Reply:
or, wait, I thought you said you were removing the `quest.previous` check entirely... I am confuzzled @.@

Reply:
I think I meant to and spaced out on it--I've made a change and am going to test it:

```
  if (user.achievements.quests
    && !user.achievements.quests[quest.prereqQuests]);
  // then if we've passed all the checks
  return quest.prereqQuests;
```

If that doesn't work, then I'll try `return false;`

Reply:
`return false;` is the way to go!

Reply:
right! `return quest.prereqQuests` would send back the whole array, when we're expecting a true/false.